nemo_skills/evaluation/metrics/base.py

```diff
@@ -29,6 +30,92 @@ def get_metrics(self):
     def reset(self):
         pass
 
+    @abc.abstractmethod
+    def get_prediction_results(self, prediction):
+        pass
+
+    def get_majority_at_k(
```

Comment:
also for majority, isn't it only applicable to math metrics? If that's the case, we can probably move this function over there and also simplify the signature (e.g. we can remove predicted_answers) and take them from predictions assuming the "predicted_answer" key is used. But if you know other use-cases that support majority, then we can keep it here